x/fungible/keeper/system_contract.go

```diff
@@ -119,11 +123,15 @@ func (k *Keeper) GetUniswapV2FactoryAddress(ctx sdk.Context) (ethcommon.Address,
 	type AddressResponse struct {
 		Value ethcommon.Address
 	}
-	var wzetaResponse AddressResponse
-	if err := sysABI.UnpackIntoInterface(&wzetaResponse, "uniswapv2FactoryAddress", res.Ret); err != nil {
+	var uniswapFactoryResponse AddressResponse
+	if err := sysABI.UnpackIntoInterface(&uniswapFactoryResponse, "uniswapv2FactoryAddress", res.Ret); err != nil {
 		return ethcommon.Address{}, cosmoserrors.Wrapf(types.ErrABIUnpack, "failed to unpack uniswapv2FactoryAddress: %s", err.Error())
 	}
-	return wzetaResponse.Value, nil
+
+	if uniswapFactoryResponse.Value == ethcommon.HexToAddress("0x0") {
```

Comment:
should we unify this check by comparing with a `ethcommon.Address{}`? I don't know which one is better but we do have many places that use `ethcommon.Address{}`